pkg/koordlet/util/system/resctrl.go

```diff
@@ -103,18 +123,30 @@ func IsSupportResctrl() (bool, error) {
 			return false, err
 		}
 		// Kernel cmdline not set resctrl features does not ensure feature must be disabled.
-		klog.Infof("IsSupportResctrl result, cpuSupport: %v, kernelSupport: %v",
+		klog.V(4).Infof("isResctrlAvailableByKernelCmd result, cpuSupport: %v, kernelSupport: %v",
 			cpuSupport, kernelCmdSupport)
 		isSupportResctrl = cpuSupport
 		isInit = true
 	}
 	return isSupportResctrl, nil
 }
 
+func IsResctrlCollectorAvailableByCpuInfo() (bool, error) {
```

Comment:
This should be checked once the agent starts and after the path flags are initialized. But the collector.Enabled() can be called for multiple times. Recommend to move the invocation inside the IsSupportResctrl.

Reply:
> This should be checked once the agent starts and after the path flags are initialized. But the collector.Enabled() can be called for multiple times. Recommend to move the invocation inside the IsSupportResctrl.

Any ideas?

Reply:
It is not about checking which flags, but when to call this check.

Reply:
> It is not about checking which flags, but when to call this check.

Ok. I will fix it soon.

Reply:
done